tests/hawkey/test_sack.cpp

```diff
@@ -83,14 +83,14 @@ START_TEST(test_list_arches)
     const char ** arches = dnf_sack_list_arches(sack);
 
     /* noarch, x86_64, athlon, i686, i586, i486, i386 */
-    fail_unless(g_strv_length((gchar**)arches) == 7);
+    fail_unless(6 <= g_strv_length((gchar**)arches));
```

Comment:
Subjective, but I don't understand why people put the constant part of a comparison check on the left. Can you please keep the original form with the constant on the right?

And again, this doesn't check for the length being 6 or 7. The array could have 23 items and the test would pass...

Reply:
Yes sure better use upper limit too. I've added it to the check and both on the right :)